File: src/cujae/inf/citi/om/data/Location.java
package cujae.inf.citi.om.data;

/* Clase que modela la ubicación geográfica de un cliente VRP a partir de sus coordenadas cartesianas*/

public class Location {

	private double axisX;
	private double axisY;
	
	public Location(){
		super();
	}

	public Location(double axisX, double axisY){
		this.axisX = axisX;
		this.axisY = axisY;
	}

	public double getAxisX() {
		return axisX;
	}

	public void setAxisX(double axisX) {
		this.axisX = axisX;
	}

	public double getAxisY() {
		return axisY;
	}

	public void setAxisY(double axisY) {
		this.axisY = axisY;
	}	
	
	/*Método que devuelve para un punto su coordenada polar Theta*/
	public double getPolarTheta(){
		return Math.atan(axisY/axisX);
	}
	
	/*Método que devuelve para un punto su coordenada polar Rho*/
	public double getPolarRho(){
		return Math.sqrt((Math.pow(axisX, 2) + Math.pow(axisY, 2)));
	}
}

File: _BHCVRP_Python_Version/cujae_inf_citi_om/_vrp_data/Location.py
import math

# Clase que modela la ubicación geográfica de un cliente VRP o depósito a partir de sus coordenadas cartesianas

class Location:
    
    def __init__(self, axis_X, axis_Y):
        self._axis_X = axis_X
        self._axis_Y = axis_Y

    def get_axisX(self):
        return self._axis_X

    def set_axisX(self, axis_X):
        self._axis_X = axis_X

    def get_axisY(self):
        return self._axis_Y

    def set_axisY(self, axis_Y):
        self._axis_Y = axis_Y

    # Método que devuelve para un punto su coordenada polar Theta 
    def get_polar_theta(self):
        return math.atan(self._axis_Y / self._axis_X)

    # Método que devuelve para un punto su coordenada Rho
    def get_polar_rho(self):
        return math.sqrt(self._axis_X ** 2 + self._axis_Y ** 2)